developer-docs-site/docs/nodes/staking.md

```diff
@@ -0,0 +1,236 @@
+
+:::tip
+In the current version of the staking on the Aptos blockchain, there is no possibility of your stake dropping below the required minimum before the lockup period expires. **This will change when Aptos implements slashing, i.e., penalty for malicious validator behavior**.
+:::
+
+### Minimum and maximum lockup period
+
+To join the validator set, you must lockup your stake for the required minimum lockup period (in seconds). You cannot withdraw any staked amount before the expiry of the lockup period. 
+
+Moreover, you can only lockup your stake for the maximum allowed period (in seconds).
+
+However, if your remaining lockup time falls below the required minimum lockup period, then you will **not** be removed from the validator set. 
+
+:::tip Note the difference
+If your staked amount falls below the required minimum you will be removed from the validator set, but if your remaining lockup time falls below the required minimum lockup period, you will not be removed from the validator set.
+:::
+
+### When the lockup period expires
+
+When your lockup period expires, you can either extend the lockup period so you can continue to validate and receive the rewards, or you can withdraw your total staked amount and stop validating.
+
+:::tip Set by the governance
+
+The above minimum and maximums for staked amount and the lockup period are decided by the Aptos governance and not by any special entity like the Aptos Labs. These minimum and maximum configurations are controlled by the covenants that the Aptos community members vote on.
+:::
+
+## Epoch
+
+An epoch in the Aptos blockchain is defined as a duration of time, in seconds, during which a number of blocks are voted on by the validators, the validator set is updated, and the rewards are distributed to the validators. 
+
+:::tip
+Currently an epoch on the Aptos blockchain is defined as sixty seconds (one hour).
+:::
+
+### Triggers at the epoch start
+
+:::tip
+See [https://github.com/aptos-labs/aptos-core/blob/0daade4f734d1ba29a896b00d7ddde2249e87970/aptos-move/framework/aptos-framework/sources/configs/stake.move#L862](https://github.com/aptos-labs/aptos-core/blob/0daade4f734d1ba29a896b00d7ddde2249e87970/aptos-move/framework/aptos-framework/sources/configs/stake.move#L862) for the full code.
+:::
+
+At the start of each epoch, the following key events are triggered:
+
+- Update the validator set by adding the pending active validators to the active validators set and by removing the pending inactive validators from the active validators set.
+- The status of the stake is updated. For example any pending active stake is updated to active stake and any pending inactive stake is updated to inactive stake.
+- The voting power of the validator in the validator set is updated.
+- Rewards are distributed to the active validators (i.e., validators in the validator set).
+- Rewards are distributed to the validators who requested to leave but have not yet been removed.
```

Comment:
can we summarize this as rewards are paid to validators that participated in the previous epoch.